tests/contrib/celery/test_integration.py

```diff
@@ -809,9 +809,26 @@ def test_thread_start_during_fork(self):
                 stderr=subprocess.STDOUT,
             )
             sleep(5)
-            celery.terminate()
+            celery.kill()
+            celery.wait()
+
+            try:
+                # Unix only. This is to avoid blocking with nothing to read.
+                import fcntl
+                import os
+
+                fd = celery.stdout.fileno()
+                fl = fcntl.fcntl(fd, fcntl.F_GETFL)
+                fcntl.fcntl(fd, fcntl.F_SETFL, fl | os.O_NONBLOCK)
```

Comment:
That's probably something that will require a second pass. In my experiments I saw the read operation hang because there was no output to read from celery.

Reply:
This commit still passes tests as far as I can tell. https://github.com/DataDog/dd-trace-py/pull/4863/commits/18609e3dba9d44f61645c36e9f9c6a3d08387023